internal/sbi/producer/ue_context_management.go

```diff
@@ -217,6 +217,22 @@ func RegistrationAmf3gppAccessProcedure(registerRequest models.Amf3GppAccessRegi
 		ue, _ := udm_context.Getself().UdmUeFindBySupi(ueID)
 		oldAmf3GppAccessRegContext = ue.Amf3GppAccessRegistration
 	}
+	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
+	// corresponding to the same (e.g. 3GPP) access, if one exists
+	if oldAmf3GppAccessRegContext != nil {
+		deregistData := models.DeregistrationData{
+			DeregReason: models.DeregistrationReason_SUBSCRIPTION_WITHDRAWN,
+			AccessType:  models.AccessType__3_GPP_ACCESS,
+		}
+		// Deregistration Notify Triggered
+		pd := callback.SendOnDeregistrationNotification(ueID,
+			oldAmf3GppAccessRegContext.DeregCallbackUri,
+			deregistData,
+		)
+		if pd != nil {
+			return nil, nil, pd
+		}
+	}
```

Comment:
```
	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
	// corresponding to the same (e.g. 3GPP) access, if one exists
	if oldAmf3GppAccessRegContext != nil {
        if ue.SameAsStoredGUAMINon3gpp(*oldAmfNon3GppRegCtx.Guami) {
            deregistData := models.DeregistrationData{
                DeregReason: models.DeregistrationReason_SUBSCRIPTION_WITHDRAWN,
                AccessType:  models.AccessType__3_GPP_ACCESS,
            }
            // Deregistration Notify Triggered
            pd := callback.SendOnDeregistrationNotification(ueID,
                oldAmf3GppAccessRegContext.DeregCallbackUri,
                deregistData,
            )
            if pd != nil {
                return nil, nil, pd
            }
        }
	}
```

Reply:
Rewrite like this:
```
	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
	// corresponding to the same (e.g. 3GPP) access, if one exists
	if oldAmf3GppAccessRegContext != nil {
		if !ue.SameAsStoredGUAMI3gpp(*oldAmfNon3GppRegCtx.Guami) {
			deregistData := models.DeregistrationData{
				DeregReason: models.DeregistrationReason_SUBSCRIPTION_WITHDRAWN,
				AccessType:  models.AccessType__3_GPP_ACCESS,
			}

			logger.UecmLog.Infof("Send DeregNotif to old AMF GUAMI=%v", oldAmf3GppAccessRegContext.Guami)
			go func() {
				pd := callback.SendOnDeregistrationNotification(ueID,
					oldAmf3GppAccessRegContext.DeregCallbackUri,
					deregistData) // Deregistration Notify Triggered
				if pd != nil {
					logger.UecmLog.Errorf("RegistrationAmf3gppAccess: send DeregNotif fail %v", pd)
				}
			}()
		}
		return nil, &registerRequest, nil
	} else {
		header = make(http.Header)
		udmUe, _ := udm_context.Getself().UdmUeFindBySupi(ueID)
		header.Set("Location", udmUe.GetLocationURI(udm_context.LocationUriAmf3GppAccessRegistration))
		return header, &registerRequest, nil
	}
```

Reply:
This handler deals with the 3GPP access case. Why do we need to check the non-3GPP access context?

Reply:
According to R15 TS 29.503 - 5.3.2.2.2 AMF registration for 3GPP access,
> 2a. On success, and if another AMF is registered for 3GPP access, the UDM updates the Amf3GppAccessRegistration resource by replacing it with the received resource information, and responds with "204 No Content". 
> UDM shall invoke the Deregistration Notification service operation towards the old AMF using the callback URI provided by the old AMF.

If the old Amf3GppAccessRegistration context exists, it should return "204 No Content" with an empty response body.

Reply:
Hi @YouShengLiu 

Please see TS 23.502 4.2.2.2.1 Step 14d:

> When the UDM stores the associated Access Type (e.g. 3GPP) together with the serving AMF as indicated in
step 14a, it will cause the UDM to initiate a Nudm_UECM_DeregistrationNotification (see clause 5.2.3.2.2) to the old AMF corresponding to the same (e.g. 3GPP) access

So, we need to check whether the access type is 3GPP or not if there is the UECM context for the same UE.
If and only if yes, UDM sends the Nudm_UECM_DeregistrationNotification to the old AMF.